### PR TITLE
Upload thumbnail on import if needed

### PR DESCRIPTION
**Pull request by pavel-savinov**

### CODE REVIEW

File: utils/get-project-content.js [hunk 1]
```diff
@@ -107,6 +107,13 @@ function _getCollectionFragments(collectionDirectory) {
           }
         };
 
+        if (metadata.thumbnailPath) {
+          metadata.thumbnailPath = path.resolve(
```
Comment (p2kmgcl) [inline utils/get-project-content.js:111]: We shouldn't put the full thumbnail path in the project content. If we need this, it should be relative to the project base (ex. `src/collection/fragment/thumbnail.jpg`

File: utils/api.js [hunk 2]
```diff
@@ -397,6 +400,65 @@ const api = {
...
+
+    const filename = `${groupId}_${fragmentEntryKey}_${path.basename(
+      thumbnailPath
+    )}`;
+
+    let fileEntry;
+
+    if (Number(previewFileEntryId) > 0) {
+      fileEntry = this.postFormData(
+        '/api/jsonws/dlapp/update-file-entry',
```
Comment (p2kmgcl) [inline utils/api.js:468]: Are these images visible in Documents and Media?

File: utils/get-project-content.js [hunk 3]
```diff
@@ -121,14 +128,9 @@ function _getCollectionFragments(collectionDirectory) {
...
-        if (
-          metadata.thumbnailPath &&
-          fs.existsSync(path.resolve(directory, metadata.thumbnailPath))
-        ) {
+        if (metadata.thumbnailPath && fs.existsSync(metadata.thumbnailPath)) {
           return Object.assign(fragment, {
-            thumbnail: fs.createReadStream(
-              path.resolve(directory, metadata.thumbnailPath)
-            )
+            thumbnail: fs.createReadStream(metadata.thumbnailPath)
```
Comment (p2kmgcl) [inline utils/get-project-content.js:133]: Storing the full ReadStream object inside projectContent case problems in the compress method, we should build a more JSON-friendly object here, or use the thumbnailPath when the image is needed

**Review by p2kmgcl — CHANGES_REQUESTED** [review]

**Comment by p2kmgcl** [comment]
@pavel-savinov can you check those problems and fix tests plz? :pray:

File: utils/get-project-content.js (hunk 3 above)
Comment (pavel-savinov) [reply utils/get-project-content.js:133]: fixed

File: utils/api.js (hunk 2 above)
Comment (pavel-savinov) [reply utils/api.js:468]: Yep, there is no way to hide it. And we have no D&M API to upload hidden files.

**Comment by p2kmgcl** [comment]
I've been discussing this with Jorge and we are gonna try another approach: finally we are going to rely on "import using zip" instead of sending everything through the API, so we do not need to maintain both API and zip imports.

I'm gonna update this process and the watch process and send them in another PR.

**Pull request closed without merging**